ludwig/hyperopt/utils.py

```diff
@@ -51,3 +54,90 @@ def should_tune_preprocessing(config):
         if f"{PREPROCESSING}." in param_name:
             return True
     return False
+
+
+def parameter_to_dict(name, value):
+    if name == ".":
+        # Parameter name ".", means top-level config
+        return value
+
+    parameter_dict = {}
+    curr_dict = parameter_dict
+    name_list = name.split(".")
+    for i, name_elem in enumerate(name_list):
+        if i == len(name_list) - 1:
+            curr_dict[name_elem] = value
+        else:
+            name_dict = curr_dict.get(name_elem, {})
+            curr_dict[name_elem] = name_dict
+            curr_dict = name_dict
+    return parameter_dict
+
+
+def feature_list_to_dict(config: Dict[str, Any]) -> Dict[str, Any]:
+    input_features_dict = {}
+    for feature in config[INPUT_FEATURES]:
+        input_features_dict[feature[NAME]] = feature
+
+    output_features_dict = {}
+    for feature in config[OUTPUT_FEATURES]:
+        output_features_dict[feature[NAME]] = feature
+
+    config = copy.copy(config)
+    config[INPUT_FEATURES] = input_features_dict
+    config[OUTPUT_FEATURES] = output_features_dict
+    return config
+
+
+def feature_dict_to_list(config: Dict[str, Any]) -> Dict[str, Any]:
+    # This works because Python dicts are order-preserving, so we do not need to
+    # do anything special to map from a key in the dict to an index in a list
```

Comment:
I guess even if these were out of order, this would work because the ordering of the features in the config doesn't make a difference to the model itself right?

Reply:
Yes, though I think preserving order if good to keep the config easy to read by the user.